File: output/ChatGPT/codenet/Java/Python/s320655609.py
import sys

class Main:
    def __init__(self):
        self.constraints = []
        self.N = 0
        self.result = 0
        self.calc()

    def calc(self):
        min_val = sys.maxsize
        for i in range(5):
            c = int(input())
            self.constraints.append(c)
            min_val = min(min_val, c)
        
        time = self.N // min_val
        if self.N % min_val != 0:
            time += 1
        time += 4
        self.result = time

if __name__ == "__main__":
    ins = Main()
    print(ins.result)

File: output/ChatGPT/codenet/Java/C++/s320655609.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Main {
    vector<long long> constraints;
    long long N;
    long long result;

public:
    Main() {
        cin >> N;
        for (int i = 0; i < 5; ++i) {
            long long c;
            cin >> c;
            constraints.push_back(c);
        }

        calc();
    }

    void calc() {
        long long minVal = *min_element(constraints.begin(), constraints.end());
        long long time = N / minVal;
        if (N % minVal != 0) {
            ++time;
        }
        time += 4;
        result = time;
    }

    void printResult() {
        cout << result << endl;
    }
};

int main() {
    Main ins;
    ins.printResult();

}